index.ts

```diff
@@ -80,14 +82,23 @@ export class SpotifyClient {
     }
   };
 
-  getLastPlayed = async () => {
+  getLastPlayed = async (limit: number = 1): Promise<ResponseTrack[]> => {
     try {
-      const accessToken = await this._genAccesToken();
-      const headers = { Authorization: `Bearer ${accessToken}` };
-      const response = await fetch(LAST_PLAYED_URL, { headers });
+      
+      if(this.accessToken === null) 
+        this.accessToken = await this._genAccesToken();
+      
+      if(limit > 50 || limit < 1)
+        throw new Error('Limit must be between 1 and 50');
+      
+      const headers = { Authorization: `Bearer ${this.accessToken.access_token}` };
+      const response = await fetch(`${LAST_PLAYED_URL}?limit=${limit}`, { headers });
       // TODO: add types once spotify-types is updated
       const responseData = (await response.json()) as any;
-      const lastPlayedTrack = filterResponse(responseData?.items[0]?.track);
+      const lastPlayedTrack = responseData.items.map((item: { track: Track; }) => {
+        const track = filterResponse(item.track);
+        return track;	
+      })
```

Comment:
We can do this I guess. Less verbose.
```ts
const lastPlayedTrack = responseData.items.map(({ track }: { track: Track }) => {
      return filterResponse(track);
 });
```

PS: We will remove typing on this map once this [issue](https://github.com/spotify-api/spotify-types/issues/4) is resolved in `spoity-types` lib

Reply:
i agree. looks more clean, dont know why i didnt do that from the start